lib/features/pay/pages/pay_who_page.dart

```diff
@@ -15,7 +16,7 @@ class PayWhoPageState extends State<PayWhoPage> {
   @override
   Widget build(BuildContext context) {
     final Map<String, dynamic> args =
-        ModalRoute.of(context)!.settings.arguments as Map<String, dynamic>;
+        GoRouterState.of(context).extra as Map<String, dynamic>;
```

Comment:
Don't pass the extras like this. Instead, pass them as the page view constructor parameters, and unpack the extras in the router file